### ISSUE TITLE
Why is the INT8 range [-127,127] instead of [-128, 127]

### ISSUE THREAD

**Issue by hy-chen** [NVIDIA/TensorRT-LLM#832]
As in this line of plugin code: 
https://github.com/NVIDIA/TensorRT-LLM/blob/main/tensorrt_llm/quantization/functional.py#L63 

> layer.get_input(0).set_dynamic_range(-127, 127)

and this line of the cuda kernel code: https://github.com/NVIDIA/TensorRT-LLM/blob/main/cpp/tensorrt_llm/kernels/layernormKernels.cu#L175

> const float dynamic_per_token_scale = 127.f / abs_max_f;

Shouldn't [-128, 127] make the model more accurate?
To utilize the range [-128,127], one could do scale = 255 / (abs_max_f * 2) instead.

**Comment by Tracin** [comment]
Yes, you are right that the range of [-128, 127] has one more fixed point than [-127, 127] and has better accuracy theoretically. However I think the improvement is minor cus 1. We already have 255 fixed points. 2. This only affects when negative value range is larger than positive.

**Comment by hy-chen** [comment]
Thanks for the reply @Tracin. Is there any downside to changing it to [-128, 127]? I am thinking of implementing this locally.

**Comment by Tracin** [comment]
I am not sure about the downside of this modification, have you verified the improvement of this changing? Like any comparison using Pytorch or something.

**Comment by chenho74** [comment]
When simulating it using Tensorflow, I observed a 1% improvement in evaluation.

**Comment by Tracin** [comment]
That is interesting, what is the model and benchmark dataset?

**Comment by nv-guomingz** [comment]
please feel free to reopen this ticket if needed.
